### PR TITLE
Add verify credential helper

### PR DESCRIPTION
**Pull request by wulfraem**
Adds helper to verify credentials to verify credentials in a convenient way.

## Details

- add helper
- add bindings for CLI, c-lib, wasm
- refactor helper & bindings align implementation style

### CODE REVIEW

File: src/helpers/credential.rs [hunk 1]
```diff
@@ -251,4 +574,121 @@ mod tests {
...
+    #[cfg(not(all(feature = "target-c-lib", feature = "capability-sdk")))]
+    async fn helper_can_verify_valid_credential() -> Result<()> {
+        let mut vade_evan = VadeEvan::new(crate::VadeEvanConfig {
+            target: DEFAULT_TARGET,
+            signer: DEFAULT_SIGNER,
+        })?;
+
+        let mut credential = Credential::new(&mut vade_evan)?;
+
+        // verify the credential issuer
```
Comment (S3bb1) [inline src/helpers/credential.rs:596]:
```suggestion
```

can this be removed?

File: src/main.rs [hunk 2]
```diff
@@ -252,6 +263,19 @@ fn add_subcommand_helper<'a>(app: App<'a, 'a>) -> Result<App<'a, 'a>> {
         } else {}
     }
 
+    cfg_if::cfg_if! {
+        if #[cfg(feature = "plugin-vc-zkp-bbs")] {
+            subcommand = subcommand.subcommand(
+                SubCommand::with_name("verify_credential")
+                    .about("Creates a `CredentialOffer` message. A `CredentialOffer` is sent by an issuer and is the response to a `CredentialProposal`. The `CredentialOffer` specifies which schema the issuer is capable and willing to use for credential issuance.")
```
Comment (S3bb1) [inline src/main.rs]: i think the `about` section is wrong :-D

File: src/main.rs [hunk 3]
```diff
@@ -252,6 +263,19 @@ fn add_subcommand_helper<'a>(app: App<'a, 'a>) -> Result<App<'a, 'a>> {
...
+                    .arg(get_clap_argument("schema_did")?)
+                    .arg(get_clap_argument("use_valid_until")?)
+                    .arg(get_clap_argument("issuer_did")?)
+                    .arg(get_clap_argument("subject_did")?),
```
Comment (S3bb1) [inline src/main.rs]: needs to be aligned as well 🍡

File: src/main.rs [hunk 4]
```diff
@@ -683,6 +707,21 @@ fn get_clap_argument(arg_name: &str) -> Result<Arg> {
...
+        "credential" => Arg::with_name("credential")
+            .long("credential")
+            .value_name("credential")
+            .help("credential to verity")
+            .takes_value(true),
+        "verification_method_id" => Arg::with_name("verification_method_id")
+            .long("verification_method_id")
+            .value_name("verification_method_id")
+            .help("ID of verification method used from issuers DID document")
+            .takes_value(true),
```
Comment (S3bb1) [inline src/main.rs]: can be removed, is unused then

File: src/main.rs [hunk 5]
```diff
@@ -198,6 +198,17 @@ async fn main() -> Result<()> {
                     )
                     .await?
             }
+            #[cfg(feature = "plugin-vc-zkp-bbs")]
+            ("verify_credential", Some(sub_m)) => {
+                get_vade_evan(sub_m)?
+                    .helper_verify_credential(
+                        get_argument_value(sub_m, "credential", None),
+                        get_argument_value(sub_m, "verification_method_id", None),
```
Comment (S3bb1) [inline src/main.rs]: unused

**Review by S3bb1 — CHANGES_REQUESTED** [review]

File: src/wasm_lib.rs [hunk 6]
```diff
@@ -194,6 +172,46 @@ cfg_if::cfg_if! {
...
+        #[cfg(feature = "plugin-vc-zkp-bbs")]
+        #[wasm_bindgen]
+        pub async fn helper_create_credential_request(
+            issuer_public_key: String,
+            bbs_secret: String,
+            credential_values: String,
+            credential_offer: String,
+            credential_schema: String
+        ) -> Result<String, JsValue> {
+
```
Comment (Tschuck) [inline src/wasm_lib.rs]:
```suggestion
```

**Comment by S3bb1** [comment]
release notes not adjusted

File: src/main.rs (hunk 3 above)
Comment (wulfraem) [reply src/main.rs]: +1 for the dango emoji 🍡 ^^

**Review by S3bb1 — APPROVED** [review]

**Review by Tschuck — APPROVED** [review]

**Pull request merged**